cmake/third_party.cmake

```diff
@@ -278,14 +278,17 @@ add_third_party(
 
 # set(Boost_DEBUG ON)
 set(Boost_USE_MULTITHREADED ON)
-SET(Boost_NO_SYSTEM_PATHS ON)
-SET(Boost_NO_BOOST_CMAKE ON)
+if(NOT USE_SYSTEM_BOOST)
+ set(Boost_NO_SYSTEM_PATHS ON)
+ set(Boost_REQUIRED_VERSION 1.68.0)
```

Comment:
I think that `set(Boost_REQUIRED_VERSION 1.68.0)` can be pulled out of `if` since you need at least this version (system or not).